js/script.js

```diff
@@ -12,21 +12,54 @@
  *       "Inspectah J"
  **/
 
+var rnd = function(array) {
+  return array[Math.floor(Math.random() * array.length)];
+}
+
+var capitalize = function(str) {
+  var ret = "";
+  for (var i = 0; i < str.length; i++) {
+    ret += str.charAt(i).toUpperCase() + ".";
+  }
+  return ret;
+}
+
 function Generator() {
 
     /* Name Arrays: Customize names to change possible output */
     this.last_names = ['the Chef', 'Digital', 'Wise', 'Knight', 'Wrecka', 'the Genius', 'the Zoo Keeper', 'the Monk', 'the Scientist', 'the Disciple', 'the Darkman', 'Pellegrino', 'the Ill Figure', 'Rocks The World', 'the Baptist',];
     this.first_names = ['Inspectah', 'Masta', 'Poppa', 'Five Foot', 'Ghostface', 'Old Dirty'];
-
 }
 
+Generator.prototype.validate = function(name) {
+  return name.match(/[A-Z][a-z]*/);
+}
 
-//Add your codez here
-
+Generator.prototype.generate = function(name) {
+  var formats =  [
+    function(name) { return rnd(this.first_names) + " " + name; },
+    function(name) { return capitalize(name) + " " + rnd(this.last_names); },
+    function(name) { return rnd(this.first_names) + " " + name + " " + rnd(this.last_names); },
+    function(name) { return name + " " + rnd(this.last_names); },
+    function(name) { return rnd(this.first_names) + " " + name.charAt(0).toUpperCase(); }
+  ];
+  var formatter = rnd(formats).bind(this);
```

Comment:
Check out using apply in this case instead of bind - this way you can do 46 and 47 in one line.